src/CMakeLists.txt

```diff
@@ -871,16 +871,14 @@ add_subdirectory(compressor)
 
 add_subdirectory(tools)
 
-# dmClock
-
-add_subdirectory(dmclock) # after gmock
-add_dependencies(tests dmclock-tests dmclock-data-struct-tests)
+# dmClock (after gmock)
 
+add_subdirectory(dmclock/src)
 if(WITH_TESTS)
-  install(PROGRAMS
-    ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/dmclock-tests
-    ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/dmclock-data-struct-tests
-    DESTINATION bin)
+  # note: add_test is not being called, so dmclock tests aren't part
+  # of ceph tests
+  add_subdirectory(dmclock/test)
```

Comment:
I understand and appreciate your efforts. If you could try out the current version of this PR it would be helpful, I think. I believe what happens is if WITH_TESTS is true, it will bring in dmclock's cmake files that deal with tests. However the tests will not be built as they're not part of any dependencies. The user would have the option of running "make dmclock-tests", for example, to build them. Does that meet your needs and not violate any established rules?

Out of curiosity, I'm interested in knowing why you do not want the tests to be built.

Also, I'm still not sure why the conditionals looking to see if gtest was a target didn't seem to work for you when you were examining the effects of the earlier PR.